src/Aspire.Dashboard/Components/Controls/LogViewer.razor.cs

```diff
@@ -1,73 +1,74 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Collections.Concurrent;
 using Aspire.Dashboard.ConsoleLogs;
 using Aspire.Dashboard.Model;
+using Aspire.Dashboard.Utils;
 using Microsoft.JSInterop;
 
 namespace Aspire.Dashboard.Components;
-public partial class LogViewer
+
+/// <summary>
+/// A log viewing UI component that shows a live view of a log, with syntax highlighting and automatic scrolling.
+/// </summary>
+public sealed partial class LogViewer
 {
-    private readonly ConcurrentQueue<IEnumerable<LogEntry>> _preRenderQueue = new();
-    private bool _renderComplete;
+    private readonly TaskCompletionSource _whenDomReady = new();
+    private readonly CancellationSeries _cts = new();
```

Comment:
`cts` is usually associated with a `CancellationTokenSource`. Could you rename this to `_cancellationSeries`? I did a double take when I saw an unexpected method call on a cts lower down.